qutip/tests/test_correlation.py

```diff
@@ -74,8 +75,8 @@ def test_correlation_solver_equivalence(solver, start, legacy):
     # broken, whereas if only one fails, then it implies that only that one is
     # broken.  We test that all solvers are equivalent by transitive equality
     # to the "me" solver.
-    base = correlation(H, start, None, times, c_ops, a.dag(), a, solver="me")
-    cmp = correlation(H, start, None, times, c_ops, a.dag(), a, solver=solver)
+    base = correlation(H, start, None, times, c_ops, x, x, solver="me")
+    cmp = correlation(H, start, None, times, c_ops, x, x, solver=solver)
```

Comment:
This isn't _really_ a sufficient regression test for this bug.  This test is called "test correlation solver equivalence", and its purpose is just to test that all the solvers give compatible answers for the particular inputs; it doesn't actually test that the answer is correct.

It would be better to leave this test as it was, and put in a new test whose purpose is specifically to catch the error you're trying to fix.  As it is, you've removed a test of the behaviour where the operators were non-Hermitian, and we still need to test that case as well.

The example given in #1460 has a simple analytic solution, so you can write a test that explicitly compares each of the solvers' ability to reproduce that - use `np.testing.assert_allclose` to compare approximate floating-point equality.  You'll have to see what tolerance is sensible, but I'd guess an absolute tolerance of around 1e-7 should be good.

I'll not say the analytic solution here in case you want to work it out yourself, but if you don't want to, then just reply to this comment and I'll tell you it.

Reply:
Oh, I just saw your comment about having modified the tests.

Yes, we should include both tests - _within reason_ the speed of tests is the secondary concern, after making sure they test everything we need them to test (note: we're not perfect at this yet).  In this case, calling any of the solvers with time-independent operators takes next-to-no time, so there's really no need to worry about it at all.

Reply:
> I'll not say the analytic solution here in case you want to work it out yourself, but if you don't want to, then just reply to this comment and I'll tell you it.

Thank you. Was good practice! :smile: